### PR TITLE
Integrate showfast export into c4PerfTest

### PR DESCRIPTION
**Pull request by borrrden**
This will allow data export to JSON that can be integrated into the showfast graphical dashboard to detect performance regressions

### CODE REVIEW

**Comment by borrrden** [comment]
I'm thinking about if this would be better to be completely opt-in via cmake options.

File: C/tests/c4PerfTest.cc [hunk 1]
```diff
@@ -152,19 +164,113 @@ class PerfTest : public C4Test {
...
+        }
+    }
+    
+    
+    string generateShowfast(Benchmark &mark, double scale, string title, const char* items = nullptr) {
+        if(isEncrypted()) {
+            title += "_encrypted";
+        }
+        
+        FLEncoder enc = FLEncoder_NewWithOptions(kFLEncodeJSON, 0, false);
```
Comment (snej) [inline C/tests/c4PerfTest.cc:179]: FYI, the Fleece C++ API (Fleece.hh) is friendlier...

**Review by snej — APPROVED** [review]
Looks fine. The Fleece C++ API would make the code cleaner, but no biggie.

File: C/tests/c4PerfTest.cc (hunk 1 above)
Comment (borrrden) [reply C/tests/c4PerfTest.cc:179]: I'll keep that in mind for an update!

**Pull request merged**